litestar/openapi/plugins.py

```diff
@@ -301,6 +303,106 @@ def render(self, request: Request, openapi_schema: dict[str, Any]) -> bytes:
+        *,
+        version: str = "1.8.0",
+        js_url: str | None = None,
+        css_url: str | None = None,
+        path: str | Sequence[str] = "/scalar",
+        **kwargs: Any,
+    ) -> None:
+        """Initialize the Scalar OpenAPI UI render plugin.
+
+        Args:
+            version: Scalar version to download from the CDN.
+                If js_url is provided, this is ignored.
+            js_url: Download url for the Scalar JS bundle.
+                If not provided, the version will be used to construct the url.
+            css_url: Download url for the Scalar CSS bundle.
+                If not provided, the Litestar-provided CSS will be used.
+            path: Path to serve the OpenAPI UI at.
+            **kwargs: Additional arguments to pass to the base class.
+        """
+        self.js_url = js_url or f"https://cdn.jsdelivr.net/npm/@scalar/api-reference@{version}"
+        self.css_url = css_url or "/schema/_litestar-internal/scalar.css"
+        super().__init__(path=path, **kwargs)
+
+    def render(self, request: Request, openapi_schema: dict[str, Any]) -> bytes:
+        """Render an HTMl page for Scalar.
+
+        .. note:: Override this method to customize the template.
+
+        Args:
+            request: The request.
+            openapi_schema: The OpenAPI schema as a dictionary.
+
+        Returns:
+            A rendered html string.
+        """
+        head = f"""
+                  <head>
+                    <title>{openapi_schema["info"]["title"]}</title>
+                    <style>{self.style}</style>
+                    <meta charset="utf-8"/>
+                    <meta name="viewport" content="width=device-width, initial-scale=1">
+                    <link rel="shortcut icon" href="{self.favicon}">
+                    <link rel="stylesheet" type="text/css" href="{self.css_url}">
+                  </head>
+                """
+
+        body = f"""
+                <noscript>
+                    Scalar requires Javascript to function. Please enable it to browse the documentation.
+                </noscript>
+                <script
+                  id="api-reference"
+                  data-url="openapi.json">
+                </script>
+                <script src="{self.js_url}" crossorigin></script>
+                """
+
+        return f"""
+                <!DOCTYPE html>
+                    <html>
+                        {head}
+                        {body}
+                    </html>
+                """.encode()
+
+    def receive_router(self, router: Router) -> None:
+        """Receive the router that serves the OpenAPI UI.
+
+        Adds a route to serve the Scalar.com OpenAPI UI CSS.
+
+        Args:
+            router: The router that serves the OpenAPI UI.
+        """
+        router.register(
+            get("/_litestar-internal/scalar.css", media_type=MediaType.CSS, sync_to_thread=False)(
+                self.render_scalar_css
+            ),
+        )
```

Comment:
I think we should only register this route if we have to, i.e., if user hasn't provided a custom css url

Reply:
[8f80bc4](https://github.com/litestar-org/litestar/pull/2906/commits/8f80bc40ececab879b7b0ec1cdea7d76df8b6ef9)